app/src/main/java/pl/droidsonroids/droidsmap/FlowManager.kt

```diff
@@ -0,0 +1,37 @@
+package pl.droidsonroids.droidsmap
+
+import pl.droidsonroids.droidsmap.feature.office.ui.OfficeUiGateway
+import pl.droidsonroids.droidsmap.feature.room.ui.RoomUiGateway
+
+class FlowManager(
+        private val officeFeatureGateway: OfficeUiGateway,
+        private val roomFeatureGateway: RoomUiGateway,
+        private val terminateCallback: TerminationCallback) : ForwardFlowChangeListener {
+
+    var currentPerspective = Perspective.OFFICE
+
+    init {
+        officeFeatureGateway.registerFlowChangeCallback(this)
```

Comment:
Jeśli register, to powinien być też unregister. A jeśli to jest na stałe, to czemu tego nie przekazać przez konstruktor?